### DISCUSSION TITLE
Using Arduino_Core_STM32 in a commercial project

### DISCUSSION THREAD

**Discussion by bayrem-gharsellaoui** [stm32duino/Arduino_Core_STM32#2316] [General]
Hello,

I want to use the Arduino_Core_STM32 (PlatformIO framework) in a commercial project.
I noticed that it uses **LGPLv2.1** licence for some parts.

my question is:
Is it possible to use the **Arduino_Core_STM32** for a commercial project without sharing my project code, knowing that I'm using it as is, without modifying any source code.

Best regards
Bayrem

**Comment by fpistm** [comment]
Hi @kaizoku-oh 
If you use the core as a library, I guess it is aligned with the LGPL. In any case, if user ask the code of the library you will have to share it.
Note that this is my understanding of the License and could not be interpreted like an authorization to use the code for your project.
You should see with your/a legal department about this usage.

Last note, we provide the software as-is without any warranty.

**Reply by bayrem-gharsellaoui** [reply]
So just to make sure I understand correctly:
if user ask the code of the library then I have to share only the library (Arduino_Core_STM32) and not my project which uses the library?

PS: In the case of PlatformIO it compiles the Arduino_Core_STM32 into a static library and later link it with the object code from my project source code (does that help with avoid sharing the code of the project?)

**Reply by fpistm** [reply]
I don't know how pio generate the final "application" so hard to tell and can't confirm this. As a first guess (if your description is correct) you have only to provide the objects of your project in order to allow end user to be able to rebuild the entire application with the modified library (Arduino_Core_STM23).

**Reply by bayrem-gharsellaoui** [reply]
But what if I'm using the library (Arduino_Core_STM32) as is, and not modified, do I still nee dto share my project or just share the fork or the original link to the Arduino_Core_STM32?

**Reply by fpistm** [reply]
As stated:
> You should see with your/a legal department about this usage.

**Reply by bayrem-gharsellaoui** [reply]
Thank you for your responses and your dedication to maintaining the Arduino_Core_STM32 library.

I appreciate your advice regarding consulting with legal departments regarding the usage of the library. However, I'd like to express some concerns regarding the licensing implications.

As the maintainer of the project, you may not have been directly involved in selecting the license for the Arduino_Core_STM32 library or fully aware of the reasons behind it. It's possible that there are nuances or considerations related to the license that we may not be aware of.

Would it be possible for you to provide more insights into the licensing choices made for the library, or perhaps direct me to any documentation or resources that explain the reasoning behind it? Understanding the license better would help me ensure that I'm in compliance with its terms and can properly share my project or modifications with the community if required.

Thank you for your understanding

**Reply by fpistm** [reply]
Third party Arduino core have to provide the implementations of the Arduino API's reference. Those definition are licensed LGPL by Arduino.
That's why the source code of those implementations have to be provided as source code. 
Some extra library are provided with compatible license and in a way it can be redistributed. All license are referenced in the core.
The license of the application (so called your project) is not the concern of the core providers. It is the responsibility of the application owners to choose the license they want while it is compatible with all software brick they used.
